app/uk/gov/hmrc/gform/validation/ValidationService.scala

```diff
@@ -192,51 +191,66 @@ class ComponentsValidator(fieldValue: FieldValue, data: Map[FieldId, Seq[String]
   private def textValidationImpl(fieldValue: FieldValue, text: Text)(data: Map[FieldId, Seq[String]]): ValidatedType = {
     val textData = data.get(fieldValue.id).toList.flatten
     (fieldValue.mandatory, textData.filterNot(_.isEmpty()), text.constraint) match {
-      case (true, Nil, _) => Invalid(Map(fieldValue.id -> Set(fieldValue.errorMessage.getOrElse("Please enter required data"))))
+      case (true, Nil, _) => Invalid(Map(fieldValue.id -> errors("Please enter required data")))
       case (_, _, AnyText) => Valid(())
       case (_, value :: Nil, Sterling) => validateNumber(value, 11, TextConstraint.defaultFactionalDigits, true)
-      case (_, value :: Nil, UkBankAccountNumber) => checkLength(value, 8)
-      case (_, value :: Nil, UkSortCode) => checkLength(value, 2)
+      case (_, value :: Nil, UkBankAccountNumber) => checkNumberLength(value, 8)
+      case (_, value :: Nil, UkSortCode) => checkNumberLength(value, 2)
+      case (_, value :: Nil, TelephoneNumber) => textValidator(value, 0, 30, true)
+      case (_, value :: Nil, Email) => email(value)
       case (_, value :: Nil, Number(maxWhole, maxFractional, _)) => validateNumber(value, maxWhole, maxFractional, false)
       case (_, value :: Nil, PositiveNumber(maxWhole, maxFractional, _)) => validateNumber(value, maxWhole, maxFractional, true)
       case (_, value :: rest, _) => Valid(()) // we don't support multiple values yet
     }
   }
 
-  private def checkLength(value: String, desiredLength: Int) = {
+  private def textValidator(value: String, min: Int, max: Int, mandatory: Boolean) =
```

Comment:
Shouldn't we also be checking for valid chars? 0-9, "ext", space, ".", "+","(", ")"